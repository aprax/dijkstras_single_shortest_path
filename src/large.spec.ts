import findShortestPath from ".";
import type { Nodes } from './types';

// global.console = require("console");

const nodes: Nodes = new Map([
	['S', [['A', 8], ['D', 2], ['G', 3]]],
	['A', [['B', 3], ['E', 7]]],
	['B', [['C', 4]]],
	// C has no edges
	['D', [['A', 2], ['E', 13]]],
	['E', [['G', 9], ['F', 10], ['C', 0]]],
	['F', [['C', 8]]],
	['G', [['F', 5], ['S', 5]]],
]);

nodes.forEach((_, key) => {
	it(`should find the shortest paths to all other nodes from ${key}`, () => {
		expect(findShortestPath(key, nodes)).toMatchSnapshot();
	})
})


