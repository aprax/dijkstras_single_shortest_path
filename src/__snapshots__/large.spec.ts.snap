// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should find the shortest paths to all other nodes from A 1`] = `
Map {
  "A" => Array [
    "A",
    0,
  ],
  "B" => Array [
    "A",
    3,
  ],
  "E" => Array [
    "A",
    7,
  ],
  "C" => Array [
    "B",
    7,
  ],
  "G" => Array [
    "E",
    16,
  ],
  "F" => Array [
    "E",
    17,
  ],
  "S" => Array [
    "G",
    14,
  ],
}
`;

exports[`should find the shortest paths to all other nodes from B 1`] = `
Map {
  "B" => Array [
    "B",
    0,
  ],
  "C" => Array [
    "B",
    4,
  ],
}
`;

exports[`should find the shortest paths to all other nodes from D 1`] = `
Map {
  "D" => Array [
    "D",
    0,
  ],
  "A" => Array [
    "D",
    2,
  ],
  "E" => Array [
    "D",
    13,
  ],
  "C" => Array [
    "E",
    13,
  ],
  "G" => Array [
    "E",
    22,
  ],
  "F" => Array [
    "E",
    23,
  ],
  "S" => Array [
    "G",
    14,
  ],
}
`;

exports[`should find the shortest paths to all other nodes from E 1`] = `
Map {
  "E" => Array [
    "E",
    0,
  ],
  "C" => Array [
    "E",
    0,
  ],
  "G" => Array [
    "E",
    9,
  ],
  "F" => Array [
    "E",
    10,
  ],
  "S" => Array [
    "G",
    14,
  ],
}
`;

exports[`should find the shortest paths to all other nodes from F 1`] = `
Map {
  "F" => Array [
    "F",
    0,
  ],
  "C" => Array [
    "F",
    8,
  ],
}
`;

exports[`should find the shortest paths to all other nodes from G 1`] = `
Map {
  "G" => Array [
    "G",
    0,
  ],
  "F" => Array [
    "G",
    5,
  ],
  "S" => Array [
    "G",
    5,
  ],
}
`;

exports[`should find the shortest paths to all other nodes from S 1`] = `
Map {
  "S" => Array [
    "S",
    0,
  ],
  "A" => Array [
    "D",
    4,
  ],
  "D" => Array [
    "S",
    2,
  ],
  "G" => Array [
    "S",
    3,
  ],
  "B" => Array [
    "A",
    11,
  ],
  "E" => Array [
    "A",
    15,
  ],
  "F" => Array [
    "G",
    8,
  ],
  "C" => Array [
    "B",
    7,
  ],
}
`;
